apps/challenges/aws_utils.py

```diff
@@ -0,0 +1,301 @@
+                    "awslogs-region": "us-east-1",
+                    "awslogs-group": "{log_group}",
+                    "awslogs-stream-prefix": "{queue_name}"  # log stream format: prefix-name/container-name/ecs-task-id
+                }},
+            }},
+            "healthCheck": {{
+                "command": [  # Need healthcheck command.
+                    ""
+                ],
+                "interval": 30,
+                "timeout": 5,
+                "retries": 3,
+                "startPeriod": 0
+            }},
+        }}
+    ],
+    "requiresCompatibilities":[
+        "FARGATE"
+    ],
+    "cpu":"",  # Need hard limits.
+    "memory":"",
+}}
+"""
+
+service_definition = """
+{{
+    "cluster":"Challenge_Cluster",
+    "serviceName":"{service_name}",
+    "taskDefinition":"{task_def_arn}",
+    "desiredCount":1,
+    "clientToken":"{client_token}",
+    "launchType":"FARGATE",
+    "platformVersion":"LATEST",
+    "networkConfiguration":{{ # Need to create VPC before filling this.
+        "awsvpcConfiguration": {{
+            "subnets": [
+                # "string",
+            ],
+            'securityGroups': [
+                "string",
+            ],
+            "assignPublicIp": "ENABLED"
+        }}
+    }},
+    "healthCheckGracePeriodSeconds":30,
+    "schedulingStrategy":"REPLICA",
+    "deploymentController":{{
+        "type": "ECS"
+    }},
+    "enableECSManagedTags":True,
+    "propagateTags":"SERVICE"
+}}
+"""
+
+update_service_args = """
+{{
+    "cluster":"Challenge_Cluster",
+    "service":"{service_name}",
+    "desiredCount":num_of_tasks,
+    "taskDefinition":"{task_def_arn}",
+    "forceNewDeployment":False
+}}
+"""
+
+aws_keys = {
+    "AWS_ACCOUNT_ID": os.environ.get("AWS_ACCOUNT_ID", "x"),
+    "AWS_ACCESS_KEY_ID": os.environ.get("AWS_ACCESS_KEY_ID", "x"),
+    "AWS_SECRET_ACCESS_KEY": os.environ.get("AWS_SECRET_ACCESS_KEY", "x"),
+    "AWS_REGION": os.environ.get("AWS_DEFAULT_REGION", "us-east-1"),
+}
+
+
+def client_token_generator():
+    client_token = ''.join(random.choices(string.ascii_letters + string.digits, k=32))
+    return client_token
+
+"""
+Called by: create_service_by_challenge_pk (below).
+
+Registers the task definition for a new challenge, and raises a custom exception
+if the worker service is already active (which is caught by the create_service custom method.).
+"""
+
+
+def register_task_def_by_challenge_pk(client, queue_name, challenge):
+    container_name = "worker_{}".format(queue_name)
+    image = "{AWS_ACCOUNT_ID}.dkr.ecr.us-east-1.amazonaws.com/evalai-production-worker:latest".format(AWS_ACCOUNT_ID=aws_keys["AWS_ACCOUNT_ID"])
+    AWS_DEFAULT_REGION = aws_keys["AWS_REGION"]
+    log_group = "{}_logs".format(queue_name)
+
+    task_role_arn = os.environ.get("TASK_ROLE_ARN", "")
+    execution_role_arn = os.environ.get("TASK_EXECUTION_ROLE_ARN", "")
+
+    definition = task_definition.format(queue_name=queue_name, task_role_arn=task_role_arn,
```

Comment:
The code after line 160 should not be executed in case the `task_role_arn` or `execution_role_arn` or any other kwargs passed in the format() is None or a blank string. Please do proper error handling here based if some configuration is missing. If we don't do that, it would be super hard to debug later. 

Please let me know if you have questions.

Reply:
Hey, sorry for the delay. Right, that would mean the worker won't have the permissions. So, I've changed it such that an appropriate error message is displayed in the admin page.